### PR TITLE
Updates WorleyNoise to be multi-threaded and faster

### PR DESCRIPTION
**Pull request by EdgeLordExe**
Another noise algorithm updated in my streak, this time it's worleynoise, since i didn't know jack shit and i  heavily abused Rc<> to acomplish my goals it was really shit. Now it is multi-threaded, blazing fast and really sexy lookin. NOTE: this update changes ABI of worley noise slightly.

obraz

### CODE REVIEW

**Comment by EdgeLordExe** [comment]
Hello could someone review this?

**Comment by AffectedArc07** [comment]
@tgstation/rust-g-dev-team

File: src/worleynoise.rs [hunk 1]
```diff
@@ -41,189 +56,198 @@ fn worley_noise(
...
+        self.reg_vec.par_iter_mut().flatten().for_each(|region| {
+            let mut rng = ThreadRng::default();
+            {
+                let mut write_guard = node_counter.write().unwrap();
+                if (*write_guard < RANGE) && rng.gen_range(0..100) > node_chance {
+                    *write_guard += 1;
+                    return;
                 }
+                *write_guard = 0;
+            }
```
Comment (ZeWaka) [inline src/worleynoise.rs:95]: what's even going on here, comment?

Comment (EdgeLordExe) [reply src/worleynoise.rs:95]: basically, to make sure the algorithm works optimally we **have to** ensure we spawn at least some nodes, this amount of nodes scales inversely to range.

Comment (ZeWaka) [reply src/worleynoise.rs:95]: like could you put that in a `// comment`

**Pull request merged**